app/controllers/api/services_controller.rb

```diff
@@ -18,6 +18,20 @@ def edit_resource(type, id, data)
       super(type, id, attributes)
     end
 
+    def add_resource_resource(type, id, data)
+      svc = resource_search(id, type, collection_class(type))
+      data['resources'].each do |resource_ref|
+        resource_type, resource_id = parse_href(resource_ref['href'])
+        resource = resource_search(resource_id, resource_type, collection_class(resource_type))
+        raise "Cannot assign #{resource_type} to #{service_ident(svc)}" unless resource.respond_to? :add_to_service
+        resource.add_to_service(svc)
```

Comment:
@jntullo Should we make a validation pass first and make sure all elements in the array respond to :add_to_service before we even attempt to even add a single one. Otherwise we would have added some and raised an error for the first one that doesn't respond to :added_to_service and we can't undo.